src/api/coupon.ts

```diff
@@ -0,0 +1,25 @@
+import { getCookie } from '@/libs/manageCookie';
+
+const BASE_URL = process.env.NEXT_PUBLIC_KEYDEUK_API_BASE_URL;
+
+export const getCoupons = async () => {
+  const token = await getCookie('accessToken');
+
+  if (!token) {
+    return null;
+  }
+
+  try {
+    const res = await fetch(`${BASE_URL}/api/v1/user/coupon`, {
+      cache: 'no-cache',
+      headers: {
+        'Cache-Control': 'no-cache',
+        Authorization: `Bearer ${token}`,
+      },
+    });
+    const { data } = await res.json();
+    return data;
+  } catch (error) {
+    throw error;
+  }
+};
```

Comment:
```suggestion
export const getCoupons = async () => {
  const token = await getCookie('accessToken');

  try {
    const res = await fetch(`${BASE_URL}/api/v1/user/coupon`, {
      cache: 'no-cache',
      headers: {
        'Cache-Control': 'no-cache',
        Authorization: `Bearer ${token}`,
      },
    });
    
    const { data } = await res.json();
    
    if (res.ok) {
        return data;
    }
    
throw new Error(data.message || '대충 에러메세지')
  } catch (error) {
    throw error;
  }
};
```
다시 생각해보니 예외처리를 이런식으로 하면 더 나을거같아요 null을 반환하는것 보다는 어떤 에러인지 던져주는 형식으로요!